.github/workflows/release-helm-chart.yml

```
@@ -1,30 +1,64 @@
-# TODO Once distributed chroma is operational, update this to update the
-# OSS helm chart we'll host. For now, just kick off the job which updates
-# hosted-chroma's version.
-
 name: 📦 Release Helm Chart
 
 on:
   push:
-    branches:
-      - main
     paths:
-      - 'k8s/**'
+      - k8s/distributed-chroma/Chart.yaml
   workflow_dispatch:
 
 jobs:
-  release:
+  detect-version-change:
+    name: Detect if version in Chart.yaml was changed
+    runs-on: ubuntu-latest
+    outputs:
+      version_changed: ${{ steps.detect-version-change.outputs.version_changed }}
+      version: ${{ steps.detect-version-change.outputs.version }}
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 2
+      - name: Detect if version field in Chart.yaml was changed
+        id: detect-version-change
+        shell: bash
+        run: |
+          current=$(git show HEAD:$file | yq ".version")
+          previous=$(git show HEAD^:$file | yq ".version")
+
+          echo "version=$current" >> $GITHUB_OUTPUT
+
+          if [ "$current" != "$previous" ]; then
+            echo "Version field in $file was changed from $previous to $current"
+            echo "version_changed=true" >> $GITHUB_OUTPUT
+          else
+            echo "Version field in $file was not changed"
+            echo "version_changed=false" >> $GITHUB_OUTPUT
+          fi
+        env:
+          file: k8s/distributed-chroma/Chart.yaml
+
+  publish-helm:
+    name: Publish Helm chart
+    needs: detect-version-change
     runs-on: ubuntu-latest
+    permissions:
+      id-token: write
+      contents: read
+    env:
+      AWS_REGION: us-east-1
+    if: ${{ needs.detect-version-change.outputs.version_changed == 'true' || github.event_name == 'workflow_dispatch' }}
     steps:
-    - name: Trigger Hosted Chroma Coordinator Release
-      uses: actions/github-script@v6
-      with:
-        github-token: ${{ secrets.HOSTED_CHROMA_WORKFLOW_DISPATCH_TOKEN }}
-        script: |
-          const result = await github.rest.actions.createWorkflowDispatch({
-            owner: 'chroma-core',
-            repo: 'hosted-chroma',
-            workflow_id: 'copy-oss-helm.yaml',
-            ref: 'main'
-          })
-          console.log(result)
+      - uses: actions/checkout@v3
+      - name: Configure AWS Credentials
+        uses: aws-actions/configure-aws-credentials@v3
+        with:
+          role-to-assume: ${{ vars.AWS_ECR_OIDC_ARN }}
+          aws-region: ${{ env.AWS_REGION }}
+      - name: Login to Amazon ECR
+        id: login-ecr
+        uses: aws-actions/amazon-ecr-login@v2
+      - name: Setup Helm
+        uses: azure/setup-helm@v4
+      - name: Package Helm chart
+        run: helm package chart
+      - name: Publish Helm chart
+        run: helm push distributed-chroma-${{ needs.detect-version-change.outputs.version }}.tgz oci://${{ vars.AWS_ECR_ACCOUNT_ID }}.dkr.ecr.${{ env.AWS_REGION }}.amazonaws.com/charts
```

I think this doesn't need the `/charts` on the end based on the docs that I was reading ([ref](https://docs.aws.amazon.com/AmazonECR/latest/userguide/push-oci-artifact.html)).
```suggestion
        run: helm push distributed-chroma-${{ needs.detect-version-change.outputs.version }}.tgz oci://${{ vars.AWS_ECR_ACCOUNT_ID }}.dkr.ecr.${{ env.AWS_REGION }}.amazonaws.com
```

---

it doesn't need it, but I figured it would be nice to namespace it
can remove if you think it's cleaner